src/Aspire.Hosting.AWS.DynamoDBLocal/IDynamoDBLocalResource.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.AWS.DynamoDBLocal;
+
+/// <summary>
+/// Represents a DynamoDB local resource. This is a dev only resources and will not be written to the project's manifest.
+/// </summary>
+public interface IDynamoDBLocalResource : IResourceWithEnvironment, IResourceWithBindings
```

Comment:
This should probably implement `IResourceWithConnectionString`.

Reply:
I don't understand how this would work because I'm not setting a connection string. The AWS SDK looks for a specific environment variable that does not go into the connection string grouping.

Reply:
Can you provide some more detail on how the environment variables work, especially in instances where you might be connecting to multiple databases.

There are two parts to consider here. How we inject the environment variables on the AppModel side, and how they are consumed on the service side.

@davidfowl this could challenge some of our assumptions about how WithReference works with connection strings. We might need another primative for things that can't use connection strings where the resource can be an _environment producer_ and be given a context to populate variables.

Reply:
Here are our docs on service specific endpoints. https://docs.aws.amazon.com/sdkref/latest/guide/feature-ss-endpoints.html. This is a feature implemented across all AWS SDKs not just the .NET SDK. 

> There are two parts to consider here. How we inject the environment variables on the AppModel side, and how they are consumed on the service side.

Using this feature of the SDK allows us to not have to implement the second part and users don't have to change any existing code to take advantage of it.

In most cases in AWS you are not "connecting" to a resource you sending request to a specific AWS region/account and the resource being acted on is a parameter to on the request. In DynamoDB's case there is no concept of a "database". An account/region just has a collection of tables making account and region the partitioning point.